docs-ref-conceptual/includes/query-azure-cli-concepts.md

```diff
@@ -0,0 +1,655 @@
+
+The following is an example of an array:
+```json
+[ 
+  1,
+  2,
+  3
+]
+```
+
+The following is an example of a dictionary:
+```json
+{
+  "isRunning": false,
+  "time": "12:00",
+  "number": 1
+}
+```
+
+Commands that _could_ return more than one object return an array, and commands that _always_ return _only_ a single object return a dictionary.
+
+## Get properties in a dictionary
+
+Working with dictionary results, you can access properties from the top level with just the key. The `.` (__subexpression__) character is used to access properties of nested dictionaries. Before introducing queries, take a look at the unmodified output of the [az vm show](/cli/azure/vm#az-vm-show) command:
+
+### [Bash](#tab/bash)
+
+```azurecli-interactive
+az vm show --resource-group QueryDemo --name TestVM
+```
+
+### [PowerShell](#tab/powershell)
+
+To avoid possible quoting issues with older versions of PowerShell, please ensure you are using the latest version. To install the latest version of PowerShell please see [Install PowerShell on Windows, Linux, and macOS](/powershell/scripting/install/installing-powershell).
+
+```powershell
+az vm show --resource-group QueryDemo --name TestVM
+```
+
+### [Cmd](#tab/cmd)
+
+```cmd
+az vm show --resource-group QueryDemo --name TestVM
+```
+
+---
+
+The command will output a dictionary. Some content has been omitted.
+
+```output
+{
+  "additionalCapabilities": null,
+  "availabilitySet": null,
+  "diagnosticsProfile": {
+    "bootDiagnostics": {
+      "enabled": true,
+      "storageUri": "https://xxxxxx.blob.core.windows.net/"
+    }
+  },
+  ...
+  "osProfile": {
+    "adminPassword": null,
+    "adminUsername": "azureuser",
+    "allowExtensionOperations": true,
+    "computerName": "TestVM",
+    "customData": null,
+    "linuxConfiguration": {
+      "disablePasswordAuthentication": true,
+      "provisionVmAgent": true,
+      "ssh": {
+        "publicKeys": [
+          {
+            "keyData": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDMobZNJTqgjWn/IB5xlilvE4Y+BMYpqkDnGRUcA0g9BYPgrGSQquCES37v2e3JmpfDPHFsaR+CPKlVr2GoVJMMHeRcMJhj50ZWq0hAnkJBhlZVWy8S7dwdGAqPyPmWM2iJDCVMVrLITAJCno47O4Ees7RCH6ku7kU86b1NOanvrNwqTHr14wtnLhgZ0gQ5GV1oLWvMEVg1YFMIgPRkTsSQKWCG5lLqQ45aU/4NMJoUxGyJTL9i8YxMavaB1Z2npfTQDQo9+womZ7SXzHaIWC858gWNl9e5UFyHDnTEDc14hKkf1CqnGJVcCJkmSfmrrHk/CkmF0ZT3whTHO1DhJTtV stramer@contoso",
+            "path": "/home/azureuser/.ssh/authorized_keys"
+          }
+        ]
+      }
+    },
+    "secrets": [],
+    "windowsConfiguration": null
+  },
+  ....
+}
+```
+
+The following command gets the SSH public keys authorized to connect to the VM by adding a query:
+
+### [Bash](#tab/bash)
+
+```azurecli-interactive
+az vm show --resource-group QueryDemo --name TestVM --query 'osProfile.linuxConfiguration.ssh.publicKeys'
+```
+
+### [PowerShell](#tab/powershell)
+
+```powershell
```

Comment:
Use `powershell-interactive` to support 'Try It` button as well? Please check other places as well.

Reply:
@jsntcy , right now the TryIt button for Azure CLI doesn't try anything.  It only launches Azure Cloud Shell, so I haven't been encouraging it's use.  However, I have a request into engineering to improve this.  Mike Robbins, the Azure PowerShell doc's lead likes the behavior of `powershell-interactive`.  Theoretically we aren't supposed to use `-interactive` unless every code block in an article belongs to a single tool.  That is why it hasn't been used in this article.

@HannahZhuSWE , when there are conceptual tabs, I think we can change the `-interactive` rule to be "when every code block on a tab belongs to the same tool", agreed?  Consequently, it would probably be a good thing to do a global find/replace and add the `-interactive` suffix.  In saying this, I am also assuming that the CLI `TryIt` button changes will get done soon, and then we won't need to circle back to update this article.

Reply:
@jsntcy @dbradish-microsoft Thank you so much for the suggestions I'll be sure to replace output with json and update the powershell code blocks to be powershell-interactive.